wcomponents-core/src/main/java/com/github/bordertech/wcomponents/AbstractWComponent.java

```diff
@@ -436,11 +437,11 @@ public final void serviceRequest(final Request request) {
 	 */
 	private static void collateVisible(final WComponent component, final List<WComponent> list) {
 		if (component.isVisible()) {
+
 			if (component instanceof Container) {
-				final int size = ((Container) component).getChildCount();
 
-				for (int i = 0; i < size; i++) {
-					collateVisible(((Container) component).getChildAt(i), list);
+				for (WComponent child : ((Container) component).getChildren()) {

```

Comment:
@jonathanaustin Thanks for the feedback. How about if getChildren returns an unmodifiable list as my PR uses but if I add another method getModifiableChildren() that passes through the List from the ComponentModel or an empty list if the Component Model's children is null. Collections.emptyList() returns a singleton instance so there is no overhead of constructing a new list.